db/migrate/20220428071221_restore_defaults_on_empty_settings.rb

```diff
@@ -0,0 +1,22 @@
+class RestoreDefaultsOnEmptySettings < ActiveRecord::Migration[6.1]
+  def up
+    Setting
+      .where(value: '')
+      .find_each do |setting|
+      definition = Settings::Definition[setting.name]
+
+      if definition.nil?
```

Comment:
I found one setting `wiki_compression` that is apparently no longer in use, but I wanted to be careful and rather log in 12.1 which settings are missing just in case we forgot any definition. We can then clean up all remnant settings in a later release.